### ISSUE TITLE
`app` query parameter should be supported (like `lang`)

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-server#16597]
The `app` query parameter isn't supported but the frontend passes it as we expect links to be configured with the right `app` (`firefox` or `android`). It isn't a problem in real life because we don't really have any difference between the two supported apps but it actually breaks Fenix installs from AMO because Fenix expects (rightly?) the download URL to contain "android" and we return download URLs with (random) apps.. Given that the frontend passes the right `app` value to the server, we only need to honor this query parameter to fix Fenix and return correct URLs.

In other words, we should probably support `app` in the query string like we do for `lang`.

**Comment by ioanarusiczki** [comment]
@willdurand Making a request with the addon detail API https://addons-dev.allizom.org/api/v5/addons/addon/gmailnoadabc@mywebber.com/?app=android or addons recommendations https://addons-dev.allizom.org/api/v5/addons/recommendations/?guid=gmailnoadabc@mywebber.com&recommended=true&lang=en-US (which I see exposed on the frontend when loading an addon from the android recommended shelf) and I pass the parameter app=android I get the following for the file download url:
"url": "https://addons-dev.allizom.org/android/downloads/file/409056/ublock_origin-2.3.4-an+fx.xpi",

If there's anything else I should be looking for let me know but I think that this should be the expected.

**Comment by willdurand** [comment]
> If there's anything else I should be looking for let me know but I think that this should be the expected.

@ioanarusiczki this issue is kinda related to Fenix so maybe you could try to verify https://github.com/mozilla/addons-frontend/issues/10100

**Comment by ioanarusiczki** [comment]
@willdurand Yes, I was expecting the fix for the frontend issue. I'll try to check it on -dev first.
